parametric/apps/golang/main.go

```diff
@@ -4,101 +4,43 @@ import (
-	var opts []tracer.StartSpanOption
-	if args.ParentId != nil && *args.ParentId > 0 {
-		parent := s.spans[*args.ParentId]
-		opts = append(opts, tracer.ChildOf(parent.Context()))
-	}
-	if args.Resource != nil {
-		opts = append(opts, tracer.ResourceName(*args.Resource))
-	}
-	if args.Service != nil {
-		opts = append(opts, tracer.ServiceName(*args.Service))
-	}
-	if args.Type != nil {
-		opts = append(opts, tracer.SpanType(*args.Type))
-	}
-	span := tracer.StartSpan(args.Name, opts...)
-
-	if args.Origin != nil {
-		span.SetTag("_dd.origin", *args.Origin)
-	}
-	s.spans[span.Context().SpanID()] = span
-	return &StartSpanReturn{
-		SpanId:  span.Context().SpanID(),
-		TraceId: span.Context().TraceID(),
-	}, nil
-}
-
-func (s *apmClientServer) SpanSetMeta(ctx context.Context, args *SpanSetMetaArgs) (*SpanSetMetaReturn, error) {
-	span := s.spans[args.SpanId]
-	span.SetTag(args.Key, args.Value)
-	return &SpanSetMetaReturn{}, nil
-}
-
-func (s *apmClientServer) SpanSetMetric(ctx context.Context, args *SpanSetMetricArgs) (*SpanSetMetricReturn, error) {
-	span := s.spans[args.SpanId]
-	span.SetTag(args.Key, args.Value)
-	return &SpanSetMetricReturn{}, nil
-}
-
-func (s *apmClientServer) FinishSpan(ctx context.Context, args *FinishSpanArgs) (*FinishSpanReturn, error) {
-	span := s.spans[args.Id]
-	span.Finish()
-	return &FinishSpanReturn{}, nil
-}
-
-func (s *apmClientServer) FlushSpans(context.Context, *FlushSpansArgs) (*FlushSpansReturn, error) {
-	tracer.Flush()
-	s.spans = make(map[uint64]tracer.Span)
-	return &FlushSpansReturn{}, nil
-}
-
-func (s *apmClientServer) FlushTraceStats(context.Context, *FlushTraceStatsArgs) (*FlushTraceStatsReturn, error) {
-	tracer.Flush()
-	s.spans = make(map[uint64]tracer.Span)
-	return &FlushTraceStatsReturn{}, nil
-}
-
-func (s *apmClientServer) StopTracer(context.Context, *StopTracerArgs) (*StopTracerReturn, error) {
-	tracer.Stop()
-	return &StopTracerReturn{}, nil
+	spans     map[uint64]tracer.Span
+	otelSpans map[uint64]ot_api.Span
+	tp        *ot.TracerProvider
+	tracer    ot_api.Tracer
 }
 
-func (s *apmClientServer) SpanSetError(ctx context.Context, args *SpanSetErrorArgs) (*SpanSetErrorReturn, error) {
-	span := s.spans[args.SpanId]
-	span.SetTag("error", true)
-	span.SetTag("error.msg", args.Message)
-	span.SetTag("error.type", args.Type)
-	span.SetTag("error.stack", args.Stack)
-	return &SpanSetErrorReturn{}, nil
+func (s *apmClientServer) StartTracer(context.Context, *StartTracerArgs) (*StartTracerReturn, error) {
+	s.tp = ot.NewTracerProvider()
+	otel.SetTracerProvider(s.tp)
+	//todo tracer options go here
+	s.tracer = s.tp.Tracer("")
+	return &StartTracerReturn{}, nil
 }
 
 func newServer() *apmClientServer {
 	s := &apmClientServer{
-		spans: make(map[uint64]tracer.Span),
+		spans:     make(map[uint64]tracer.Span),
+		otelSpans: make(map[uint64]ot_api.Span),
 	}
 	return s
 }
 
 func main() {
-	tracer.Start()
-	defer tracer.Stop()
```

Comment:
Make a note to make sure to call the tracer.Start as before it was automatically invoked in the main func, and now in conftest.py